pkg/trace/api/otlp.go

```diff
@@ -475,7 +498,9 @@ func setMetricOTLP(s *pb.Span, k string, v float64) {
 
 // convertSpan converts the span in to a Datadog span, and uses the rattr resource tags and the lib instrumentation
 // library attributes to further augment it.
-func (o *OTLPReceiver) convertSpan(rattr map[string]string, lib pcommon.InstrumentationScope, in ptrace.Span) *pb.Span {
+//
+// ctags will be used to write container tags to. Existing ones are not overridden.
```

Comment:
This means the first container tag value always wins, but OTel spans can be in any order (AFAIK), will that cause any problem? Or is it guaranteed all spans in a ResourceSpans payload have the same tag value?

Reply:
> is it guaranteed all spans in a ResourceSpans payload have the same tag value?

I don't think we have any such guarantee

Reply:
It doesn't matter. Container tags are per container and per resource. There can't be two different spans coming from the same resource having the same container tag key with two different values.

Reply:
Thanks, that makes sense to me

Reply:
I hope it does, it's more like "best effort" attempt to extract the metadata. I agree the algorithm is not perfect, but it should work correctly if everything is set up right. And if it's not set up right -- it's hard to tell what values to choose anyway so I'm not sure how we could optimise it.